mesa/experimental/cell_space/cell_collection.py

```diff
@@ -48,8 +48,10 @@ def __init__(
         else:
             self._cells = {cell: cell.agents for cell in cells}
 
-        #
-        self._capacity: int = next(iter(self._cells.keys())).capacity
+        # Get capacity from first cell if collection is not empty
+        self._capacity: int | None = (
+            next(iter(self._cells.keys())).capacity if self._cells else None
```

Comment:
I am fine with doing it this way, but the pythonic way is to do a try catch instead (its beter to ask forgiveness then permission).

Reply:
I think a try-catch is overkill here, since there are two clear, predictable scenarios.